cdc/log.cc

```diff
@@ -767,64 +854,21 @@ class transformer final : public change_processor {
 
     std::vector<mutation> _result_mutations;
 
-    // Keeps the next cdc$batch_seq_no value
-    int _batch_no = -1;
-    // The timestamp of changes being currently processed
-    api::timestamp_type _ts;
-    // The cdc$time value of changes being currently processed
-    bytes _tuuid;
+    std::unique_ptr<log_mutation_builder> _builder;
```

Comment:
Question about the lifetime of this _builder. It holds a bunch of references, including one for a schema. How do we guarantee that all of these references remain valid? Is the "transformer" parent object holding both a _builder and the objects it points to (like schema_ptr and the various keys etc.)?

Reply:
Can't guarantee it in compile-time (it's not Rust unfortunately), but yes, the idea is that `transformer` keeps those things alive at least as long as `_builder`.

Reply:
I see one potential problem: if someone moves the `transformer` after constructing the `builder`, some references might get invalidated.

Although now that I think of it, none of them would:
- schemas are kept alive by other entities
- column definitions refer to things inside the schemas
- _log_mut does refer to an element in a vector from `transformer`, but the `transformer` doesn't hold the data, only a pointer to the data.

So we're double lucky:
- `transformer` isn't moved after `begin_timestamp()` is called
- even if it would, we don't strictly hold a reference to anything that is kept alive by transformer

Reply:
Anyway, I added a comment above `begin_timestamp()` that `transformer` should not be moved after calling it.

It would be better to delete `transformer`'s move constructor/assignment, but it's used once in `augment_mutation_call`. Removing the move would require taking `pre_image_select` out of transformer, which I think should be done, but maybe not in this PR.